### ISSUE TITLE
wgpu error: Validation Error - Dimension X is zero

### ISSUE THREAD

**Issue by MTRNord** [bevyengine/bevy#1664]
## Bevy version

079b3ade89195ef1aa4bcf1f8c0740a6572b84b6 

## Operating system & version

Windows 10

## What you did

The game had its window in the background. This happened randomly

## What actually happened

It caused a panic and crash

## Additional information

I am not aware of any way to reproduce it currently. Also I guess due to limited knowledge (first time user as of writing this) I am very much able to miss something.

Crash log:

```
wgpu error: Validation Error

Caused by:
    In Device::create_texture
    Dimension X is zero


thread 'main' panicked at 'Handling wgpu errors as fatal by default', C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.0\src\backend\direct.rs:1896:5
stack backtrace:
   0: std::panicking::begin_panic<str>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:519
   1: wgpu::backend::direct::default_error_handler
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.0\src\backend\direct.rs:1896
   2: core::ops::function::Fn::call<fn(wgpu::Error),tuple<wgpu::Error>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:70
   3: alloc::boxed::{{impl}}::call<tuple<wgpu::Error>,UncapturedErrorHandler,alloc::alloc::Global>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1560
   4: wgpu::backend::direct::ErrorSinkRaw::handle_error
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.0\src\backend\direct.rs:1883
   5: wgpu::backend::direct::Context::handle_error<wgpu_core::resource::CreateTextureError>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.0\src\backend\direct.rs:109
   6: wgpu::backend::direct::{{impl}}::device_create_texture
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.0\src\backend\direct.rs:1132
   7: wgpu::Device::create_texture
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.0\src\lib.rs:1591
   8: bevy_wgpu::renderer::wgpu_render_resource_context::{{impl}}::create_texture
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_wgpu\src\renderer\wgpu_render_resource_context.rs:258
   9: bevy_render::render_graph::nodes::window_texture_node::{{impl}}::update
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_render\src\render_graph\nodes\window_texture_node.rs:73
  10: bevy_wgpu::renderer::wgpu_render_graph_executor::WgpuRenderGraphExecutor::execute
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_wgpu\src\renderer\wgpu_render_graph_executor.rs:73
  11: bevy_wgpu::wgpu_renderer::WgpuRenderer::run_graph
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_wgpu\src\wgpu_renderer.rs:113
  12: bevy_wgpu::wgpu_renderer::WgpuRenderer::update
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_wgpu\src\wgpu_renderer.rs:124
  13: bevy_wgpu::get_wgpu_render_system::{{closure}}
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_wgpu\src\lib.rs:107
  14: alloc::boxed::{{impl}}::call_mut<tuple<mut bevy_ecs::core::world::World*, mut bevy_ecs::resource::resources::Resources*>,FnMut<tuple<mut bevy_ecs::core::world::World*, mut bevy_ecs::resource::resources::Resources*>>,alloc::alloc::Global>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1553
  15: bevy_ecs::system::exclusive_system::{{impl}}::run
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_ecs\src\system\exclusive_system.rs:31
  16: bevy_ecs::schedule::stage::{{impl}}::run
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_ecs\src\schedule\stage.rs:580
  17: bevy_ecs::schedule::Schedule::run_once
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_ecs\src\schedule\mod.rs:198
  18: bevy_ecs::schedule::{{impl}}::run
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_ecs\src\schedule\mod.rs:209
  19: bevy_app::app::App::update
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_app\src\app.rs:56
  20: bevy_winit::winit_runner_with::{{closure}}
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_winit\src\lib.rs:486
  21: winit::platform_impl::platform::event_loop::{{impl}}::run_return::{{closure}}<tuple<>,closure-1>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:203
  22: alloc::boxed::{{impl}}::call_mut<tuple<winit::event::Event<tuple<>>, mut winit::event_loop::ControlFlow*>,FnMut<tuple<winit::event::Event<tuple<>>, mut winit::event_loop::ControlFlow*>>,alloc::alloc::Global>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1553
  23: winit::platform_impl::platform::event_loop::runner::{{impl}}::call_event_handler::{{closure}}<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:245
  24: std::panic::{{impl}}::call_once<tuple<>,closure-0>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:344
  25: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,tuple<>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:379
  26: winit::platform_impl::platform::monitor::{{impl}}::fmt
  27: std::panicking::try<tuple<>,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:343
  28: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,tuple<>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:431
  29: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<tuple<>>::catch_unwind<tuple<>,tuple<>,closure-0>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:152
  30: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<tuple<>>::call_event_handler<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:239
  31: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<tuple<>>::move_state_to<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:341
  32: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<tuple<>>::main_events_cleared<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:227
  33: winit::platform_impl::platform::event_loop::flush_paint_messages<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:676
  34: winit::platform_impl::platform::event_loop::thread_event_target_callback::{{closure}}<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:1967
  35: std::panic::{{impl}}::call_once<isize,closure-0>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:344
  36: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:379
  37: winit::platform_impl::platform::monitor::{{impl}}::fmt
  38: std::panicking::try<isize,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:343
  39: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:431
  40: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<tuple<>>::catch_unwind<tuple<>,isize,closure-0>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:152
  41: winit::platform_impl::platform::event_loop::thread_event_target_callback<tuple<>>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:2151
  42: DefSubclassProc
  43: DefSubclassProc
  44: CallWindowProcW
  45: DispatchMessageW
  46: SendMessageTimeoutW
  47: KiUserCallbackDispatcher
  48: NtUserDispatchMessage
  49: DispatchMessageW
  50: winit::platform_impl::platform::event_loop::EventLoop<tuple<>>::run_return<tuple<>,closure-1>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:218
  51: winit::platform_impl::platform::event_loop::EventLoop<tuple<>>::run<tuple<>,closure-1>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:188
  52: winit::event_loop::EventLoop<tuple<>>::run<tuple<>,closure-1>
             at C:\Users\marce\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\event_loop.rs:154
  53: bevy_winit::run<closure-1>
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_winit\src\lib.rs:170
  54: bevy_winit::winit_runner_with
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_winit\src\lib.rs:494
  55: bevy_winit::winit_runner
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_winit\src\lib.rs:210
  56: core::ops::function::Fn::call<fn(bevy_app::app::App),tuple<bevy_app::app::App>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:70
  57: alloc::boxed::{{impl}}::call<tuple<bevy_app::app::App>,Fn<tuple<bevy_app::app::App>>,alloc::alloc::Global>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1560
  58: bevy_app::app::App::run
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_app\src\app.rs:66
  59: bevy_app::app_builder::AppBuilder::run
             at C:\Users\marce\.cargo\git\checkouts\bevy-f7ffde730c324c74\079b3ad\crates\bevy_app\src\app_builder.rs:50
  60: two_d_world_bevy::main
             at .\src\main.rs:19
  61: core::ops::function::FnOnce::call_once<fn(),tuple<>>
             at C:\Users\marce\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Mar 16 00:57:18.605 ERROR gpu_descriptor::allocator: `DescriptorAllocator` is dropped while some descriptor sets were not deallocated
```

Code can be found at https://github.com/MTRNord/bevy_learning/tree/f06ca8e721aaf0591b561b7225026a03add7faaf

**Comment by MTRNord** [comment]
Just as a note: This also can apparently happen when trying to close the window.

**Comment by atomicbitty** [comment]
This happened to me when resizing a window to zero in some dimension, should not be hard to fix, maybe I'll make a PR about that.

**Comment by mockersf** [comment]
resizing to 0 should have been fixed by #1409

**Comment by NathanSWard** [comment]
resizing to 0 should've been fixed by #1409
This also appears to be sort of a duplicate to 
#170 #899 (and a few others)
